docs/unit4.md

```diff
@@ -2,4 +2,311 @@
+
+```css
+p {
+	font-family: 'Arial';
+	color: red;
+}
+```
+
+There are three main ways of selecting HTML to style: Element, ID, and Class. Remember, ID and class can be set as attributes for each element. There are also more, base don attribute, or pseudo-classes. 
+
+```css
+p { /* Select by element tag */
+	font-family: 'Arial';
+	color: red;
+}
+.emphasize { /* Select by element class */
+	font-family: 'Arial';
+	color: red;
+}
+
+#redtext { /* Select by element id */
+	font-family: 'Arial';
+	color: red;
+}
+
+img[src] { /* Select img with src attribute */
+	width: 500px;
+}
+
+a:hover { /* When 'a' tags are hovered by your mouse they become blue */
+	color: blue
+}
+```
+
+### CSS Reference
+
+[CSS reference](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference)
+
+### Specifying Styles
+
+The best practice way of specifying styles is to create a separate `<name>.css` file, where you will put all your styles and rules. 
+
+```html
+<head>
+	 <!-- Use the './css/style.css' file for styling this document -->
+	<link rel="stylesheet" href="./css/style.css">
+</head>
+<body>
+	...
+</body>
+```
+
+### The Box Model 🔲
+
+CSS is all about boxes, in how the entire page is laid out. Each box has multiple properties in how they take up space: 
+
+- `padding` - the space around the content. In the example below, it is the space around the paragraph text.
+- `border` - the solid line that is just outside the padding.
+- `margin` - the space around the outside of the border. This is used to separate elements from its neighbors
+- `width` - the width of the content
+- `height` - the height of the content
+
+Padding, border and margin have direction properties. (e.g., padding-top,
+margin-right, border-left). 
+
+Border can have border-color, border-width, and border-style.
+
+![Borders](https://images.unsplash.com/photo-1548546738-8509cb246ed3?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=987&q=80)
+
+### Units
+
+For the above properties, they all can be set with certain units. 
+
+| Unit | Description |
+| --- | --- |
+| Pixels (px) | 1 pixel |
+| Element units (em) | Relative to the font-size of the element.  |
+| Percentages (%) | Relative to the parent element. |
+| Real-world units (in, cm) | An inch, or centimeter. Not recommended to use. |
+| Viewport (vh, vw) | Relative to 1% of the width/height of the viewport, or display. (100vh = 100% of the height of the viewport. |
+- Try to use relative units (em and %) whenever possible. Helps for accessibility, when users change default sizes.
+- Don’t use real-world units, as these are fixed regardless of device.
+
+### Display
+
+The display property is one of the most important CSS properties for controlling layout. Focusing on the two main display property values. 
+
+- `inline` - Does not start on a newline and only takes up as much width as necessary.
+    - Examples of inline-level elements: <span>, <a>, <img>
```

Comment:
```suggestion
    - Examples of inline-level elements: `<span>`, `<a>`, `<img>`
```
Wrap all of these in code tags because otherwise markdown hides these